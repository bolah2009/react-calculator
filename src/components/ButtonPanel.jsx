import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';
import styles from '../css/modules/ButtonPanel.module.css';

const buttonsLabel = [
  { group: ['AC', '+/-', '%', 'รท'], id: 'group-one' },
  { group: ['7', '8', '9', 'X'], id: 'group-two' },
  { group: ['4', '5', '6', '-'], id: 'group-three' },
  { group: ['1', '2', '3', '+'], id: 'group-four' },
  { group: ['0', '.', '='], id: 'group-five' },
];

const ButtonGroup = ({ group, id }) => {
  const buttonGroups = group.map((label, index) => (

    <Button
      key={label}
      color={index < group.length - 1 ? 'lightgrey' : 'orange'}
      name={label}
      isWide={label === '0'}
    />
  ));
  return (
    <div className={`d-flex as-c ${styles['button-group']} ${id}`}>{buttonGroups}</div>
  );
};

const ButtonPanel = () => (
  buttonsLabel.map(({ group, id }) => (
    <ButtonGroup key={id} group={group} id={id} />
  ))
);

ButtonGroup.propTypes = {
  group: PropTypes.arrayOf(PropTypes.string).isRequired,
  id: PropTypes.string.isRequired,
};

export default ButtonPanel;
