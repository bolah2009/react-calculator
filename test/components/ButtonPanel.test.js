import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../../src/components/ButtonPanel';

describe('ButtonPanel component should ', () => {
  const handleClick = jest.fn();
  test('should render buttons in group', () => {
    const buttonPanel = renderer.create(<ButtonPanel clickHandler={i => handleClick(i)} />);
    expect(buttonPanel.toJSON()).toMatchSnapshot();
  });
});
