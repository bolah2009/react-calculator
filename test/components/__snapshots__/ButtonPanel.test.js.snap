// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel component should  should render buttons in group 1`] = `
Array [
  <div
    className="d-flex as-c button-group group-one"
  >
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      %
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      รท
    </button>
  </div>,
  <div
    className="d-flex as-c button-group group-two"
  >
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      7
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      8
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      9
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      X
    </button>
  </div>,
  <div
    className="d-flex as-c button-group group-three"
  >
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      4
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      5
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      6
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      -
    </button>
  </div>,
  <div
    className="d-flex as-c button-group group-four"
  >
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      1
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      2
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      3
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      +
    </button>
  </div>,
  <div
    className="d-flex as-c button-group group-five"
  >
    <button
      className="button wide"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      0
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      .
    </button>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      =
    </button>
  </div>,
]
`;
